bigchaindb-on-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,246 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "admin_username": {
+            "defaultValue": null,
+            "type": "String",
+            "metadata": {
+                "description": "The admin username"
+            }
+        },
+        "admin_password": {
+            "defaultValue": null,
+            "type": "SecureString",
+            "metadata": {
+                "description": "The admin password"
+            }
+        },
+        "virtual_machine_name": {
+            "defaultValue": "",
+            "type": "String",
+            "metadata": {
+                "description": "A name for the virtual machine"
+            }
+        },
+        "virtual_machine_size": {
+            "defaultValue": "Standard_D2",
+            "type": "String",
+            "metadata": {
+                "description": "The virtual machine size"
+            },
+            "allowedValues": [
+                "Standard_D1",
+                "Standard_D2",
+                "Standard_D3",
+                "Standard_D4",
+                "Standard_DS1",
+                "Standard_DS2",
+                "Standard_DS3",
+                "Standard_DS4"
+            ]
+        }
+    },
+    "variables": {
+        "location": "[resourceGroup().location]",
+        "vmSize": "Standard_D2",
+        "storageAccount": "[parameters('virtual_machine_name')]",
+        "nicName": "[parameters('virtual_machine_name')]",

```

Comment:
this naming schema won't work well in a number of cases, the naming requirements for resources is different between some resources.  fine to use a prefix, but this needs to consider constraints for each name.  If you don't care about the resource names (which is perfectly fine) then just hard code the names that are not globally unique and generate ones that are - there are many samples that follow this pattern: https://github.com/Azure/azure-quickstart-templates/tree/master/101-vm-simple-windows